File: src/java/com/distributedlife/mahjong/reference/permute/SingleTilePermutator.java
package com.distributedlife.mahjong.reference.permute;

import java.util.Arrays;

public class SingleTilePermutator extends MultiplePermutator {
    public SingleTilePermutator(String tile, String suit) {
        super(Arrays.asList(tile), 1, suit);
    }
}

File: lib/permute/single-tile-permutator.js
/**
 * SingleTilePermutator - Handles permutations for single tiles
 */
import { MultiplePermutator } from './multiple-permutator.js';

export class SingleTilePermutator extends MultiplePermutator {
  constructor(tile, suit) {
    super([tile], 1, suit);
  }
}
